src/Lean/Elab/MacroArgUtil.lean

```diff
@@ -63,7 +62,18 @@ where
           else
             throwError "unknown parser declaration/category/alias '{id}'"
       pure ⟨Syntax.mkAntiquotNode kind term⟩
-    | stx, term =>
+    | stx, term => do
+      -- can't match against `` `(stx| ($stxs*)) `` as `*` is interpreted as the `stx` operator
+      if stx.raw.isOfKind ``Parser.Syntax.paren then
+        -- translate argument `v:(p1 ... pn)` where all but one `pi` produce zero nodes to
+        -- `v:pi` using that single `pi`
+        let nonNullaryNodes ← stx.raw[1].getArgs.filterM fun
+          | `(stx| $id:ident$[:$_]?) | `(stx| $id:ident($_)) => do
+            let info ← Parser.getParserAliasInfo id.getId
+            return info.stackSz? != some 0
```

Comment:
This does not cover recursive cases such as `(group(ppSpace ppSpace) ident)`, but I couldn't see any practical use cases there...